### DISCUSSION TITLE
Appender and attached databases / temporary tables

### DISCUSSION THREAD

**Discussion by stromnov** [marcboeker/go-duckdb#314] [Q&A]
Hi! Is it possible to create an Appender for a temporary table? Or is it possible to create an Appender for a specific table in an attached database?

**Comment by taniabogatsch** [comment]
1. I know it is possible to append to temporary tables in the C++ API. Could you give it a try in go-duckdb?
2. That is currently not supported, as we cannot pass the catalog (database) when creating an appender. It will become possible once these changes are released: https://github.com/duckdb/duckdb/pull/14256

**Reply by stromnov** [reply]
A snippet from the documentation says that _"temporary tables are part of the temp.main schema"_. Therefore, if I understand correctly, this is not currently possible. At least all my experiments ended with a message indicating an error _"database/sql/driver: could not create appender: duckdb error: Table "tempname" could not be found"_.

**Reply by taniabogatsch** [reply]
In that case, you could check if it succeeds with the C API on duckdb's latest `main`. If so, we need to add the respective support here after the next release. Otherwise, we'd have to add this to duckdb first. At least I don't see a reason against supporting appends to temporary tables. 🤔

**Reply by jorinvo** [reply]
I had a similar issue: I couldn't figure out how to append to a table in the non-default namespace. 

Maybe the issue for temporary tables is that you cannot set the `temp.` prefix?

**Reply by taniabogatsch** [reply]
`func NewAppender(driverConn driver.Conn, catalog, schema, table string) (*Appender, error)`

It should now be possible to create an `Appender` on a non-default catalog (namespace). Does that work?

**Reply by jorinvo** [reply]
Oh I didn't realize that I am not on the latest version. `catalog` is a great addition! That will work great for me. Thank you!

Appending to a temporary table might work in theory. But appenders use `driver.Conn` and querying uses `sql.Conn`. So not sure how to actually use the same connection in practice. But I am also not sure that appending to temporary tables is actually needed by anyone.

**Reply by taniabogatsch** [reply]
You should be able to access the raw connection:
```Go
	err = conn.Raw(func(driverConn interface{}) error {
		duckdbConn := driverConn.(*duckdb.Conn)
```

**Reply by jorinvo** [reply]
Thanks for sharing! That will be useful if someone needs this and comes across this discussion!

**Comment by mlafeldt** [comment]
👋 Thanks for starting this discussion!

This repository has moved to https://github.com/duckdb/duckdb-go

If this topic is still relevant, please continue the conversation at the new location. I'm closing this now as part of the migration.

Please see https://github.com/marcboeker/go-duckdb/issues/565 for more details.
